synapse/storage/room.py

```diff
@@ -638,3 +745,82 @@ def _get_media_mxcs_in_room_txn(self, txn, room_id):
                             remote_media_mxcs.append((hostname, media_id))
 
         return local_media_mxcs, remote_media_mxcs
+
+    @defer.inlineCallbacks
+    def get_rooms_for_retention_period_in_range(self, min_ts, max_ts, include_null=False):
+        """Retrieves all of the rooms for which a retention policy is defined with a
+        max_lifetime within ]min_ts ; max_ts]. Optionnally includes the rooms which don't
+        have a retention policy.
+
+        Args:
+            min_ts (int): Timestamp in milliseconds that define the lower limit of the
+                range to handle.
+            max_ts (int): Timestamp in milliseconds that define the higher limit of the
+                range to handle.
+            include_null (bool): Whether to include rooms which retention policy is NULL
+                in the returned set.
+
+        Returns:
+            set[dict[str, int, int]]: The rooms within this range, along with their
```

Comment:
I think it returns a list, not a set

Reply:
also: it's not a `dict[str, int, int]` (which doesn't mean anything): it's a `dict[str, \<something>]`, or as we normally write at this point, just `dict`. 

It would be helpful to list (all) the keys in the dicts.